tests/tests/test_mtls.py

```diff
@@ -0,0 +1,323 @@
+# Copyright 2020 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        https://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import json
+import os.path
+import pytest
+import shutil
+import subprocess
+import tempfile
+import time
+
+from testutils.common import create_org
+from testutils.infra.container_manager import factory
+from testutils.infra.device import MenderDevice
+
+from .. import conftest
+from ..MenderAPI import logger, reset_mender_api, auth, deploy, devauth
+from .mendertesting import MenderTesting
+
+container_factory = factory.get_factory()
+
+
+@pytest.fixture(scope="function")
+def setup_ent_mtls(request):
+    env = container_factory.getMTLSSetup()
+    request.addfinalizer(env.teardown)
+    env.setup()
+
+    mtls_username = "mtls@mender.io"
+    mtls_password = "correcthorsebatterystaple"
+
+    env.tenant = create_org(
+        "Mender",
+        mtls_username,
+        mtls_password,
+        containers_namespace=env.name,
+        container_manager=env,
+    )
+    env.user = env.tenant.users[0]
+    env.start_mtls_ambassador()
+
+    reset_mender_api(env)
+
+    auth.username = mtls_username
+    auth.password = mtls_password
+    auth.multitenancy = True
+    auth.current_tenant = env.tenant
+
+    return env
+
+
+def make_script_artifact(artifact_name, device_type, output_path):
+    script = b"""\
+#! /bin/bash
+
+set -xe
+
+# Just give it a little bit of time
+sleep 6s
+
+# Successful update
+exit 0
+"""
+
+    with tempfile.NamedTemporaryFile(suffix="testdeployment") as tf:
+        tf.write(script)
+        tf.seek(0)
+        out = tf.read()
+        logger.info(f"Script: {out}")
+        script_path = tf.name
+
+        cmd = f"mender-artifact write module-image -T script -n {artifact_name} -t {device_type} -o {output_path} -f {script_path}"
+        logger.info(f"Executing command: {cmd}")
+        subprocess.check_call(cmd, shell=True)
+
+        return output_path
+
+
+class TestClientMTLSEnterprise:
+    @MenderTesting.fast
+    @pytest.mark.parametrize("algorithm", ["rsa", "ec256", "ed25519",])
+    def test_mtls_enterprise(self, setup_ent_mtls, algorithm):
+        tenant = setup_ent_mtls.tenant
+
+        # stop the api gateway
+        setup_ent_mtls.stop_api_gateway()
+
+        # start a new mender client
+        setup_ent_mtls.new_mtls_client("mender-client", tenant.tenant_token)
+
+        # stop the Mender client
+        device = MenderDevice(setup_ent_mtls.get_mender_clients()[0])
```

Comment:
For the test framework to find the `MenderDevice`, the object needs to be referenced from the setup:

```suggestion
        setup_ent_mtls.device = MenderDevice(setup_ent_mtls.get_mender_clients()[0])
```

Reply:
Done!